game_management/game_input_handler.py

```diff
@@ -0,0 +1,470 @@
+import random
+import typing
+from random import random
+
+import pyxel
+
+from board import Board
+from calculator import clamp, complete_construction, attack_setl
+from catalogue import get_available_improvements, get_available_blessings, get_available_unit_plans, PROJECTS
+from game_management.game_controller import GameController
+from game_management.game_state import GameState
+from menu import MainMenuOption, SetupOption, WikiOption
+from models import Construction, OngoingBlessing, CompletedConstruction, Heathen, GameConfig, OverlayType, Faction, \
+    ConstructionMenu, Project
+from movemaker import set_player_construction
+from overlay import SettlementAttackType, PauseOption
+from saving.game_save_manager import GameSaveManager
+
+
+class GameInputHandler:
```

Comment:
I think there's probably no point in making this a class if all the methods are static anyway, would probably be fine to just have each of the methods as top-level functions in here.

Reply:
Probably a Java habit crossing over 😂